### PR TITLE
Fixed #26648 -- Added a system check for related_query_name.

### PR DESCRIPTION
**Pull request by koniiiik**
https://code.djangoproject.com/ticket/26648

### CODE REVIEW

File: django/db/models/fields/related.py [hunk 1]
```diff
@@ -147,6 +148,34 @@ def _check_related_name_is_valid(self):
...
+            errors.append(
+                checks.Error(
+                    "Reverse query name for '%s' ('%s') ends with an underscore." % (field_name, rel_query_name),
+                    hint=("Add or change a related_name or related_query_name "
+                          "argument to the definition for field '%s'.") % (field_name,),
+                    obj=self,
+                    id='fields.E307',
+                )
+            )
+        if '__' in rel_query_name:
```
Comment (charettes) [inline django/db/models/fields/related.py]: Use `django.db.models.constants.LOOKUP_SEP`.

**Comment by koniiiik** [comment]
Ah, just noticed I'll also need to bump the error numbers; will do tomorrow.

**Comment by phildini** [comment]
This should be documented in the 1.11 release notes.

**Comment by koniiiik** [comment]
Good point, I'll do that as soon as I get to my computer.

**Comment by timgraham** [comment]
We haven't historically included system check additions in the release notes.

merged in 686a593aaadb1c2e2053fee8c401476dcb4617a5, thanks!

**Pull request closed without merging**